Helper/FeatureSwitch/Manager.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Helper\FeatureSwitch;
+
+
+use Magento\Framework\App\Helper\Context;
+use Bolt\Boltpay\Helper\GraphQL\Client as GQL;
+use Bolt\Boltpay\Model\FeatureSwitchRepository;
+use Magento\Framework\App\Helper\AbstractHelper;
+
+class Manager extends AbstractHelper {
+
+    /**
+     * @var GQL
+     */
+    private $gql;
+
+    /**
+     * @var fsRepo
+     */
+    private $fsRepo;
+
+    /**
+     * @param Context $context
+     * @param GQL $gql
+     * @param FeatureSwitchRepository $fsRepo
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        Context $context,
+        GQL $gql,
+        FeatureSwitchRepository $fsRepo
+    ) {
+        parent::__construct($context);
+        $this->gql = $gql;
+        $this->fsRepo = $fsRepo;
+    }
+
+    /**
+     * This method gets feature switches from Bolt and updates the local DB with
+     * the latest values. To be used in upgrade data and webhooks.
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function updateSwitchesFromBolt() {
+        $switchesResponse = $this->gql->getFeatureSwitches();
+
+        $switches = array();
+        if ($switchesResponse->getData()) {
+            $data = $switchesResponse->getData();
+            if (isset($data["response"])
+                && isset($data["response"]->data)
+                && isset($data["response"]->data->plugin)
+                && isset($data["response"]->data->plugin->features)
+            ) {
+                $switches = $data["response"]->data->plugin->features;
+            }
+        }
+
+        if (!is_array($switches) || count($switches) == 0) {
+            return;
+        }
+
+        foreach ($switches as $fs) {
+            if (isset($fs->name)) {
```

Comment:
While this check is safe, why have it? All 4 values are necessary for `upsertByName`, right? When would a name be missing but not the other 3 values?

Reply:
TBH probably overkill. I am trying hard to make sure the code will not barf by checking everything. however value and default value are booleans so unset they are false. that said i could also check roll out %age.

let me know what you decide

Reply:
That's prooobably fine, and I suppose if name is present then %age likely will too. I thought raising this to get your thoughts was worth it, but this doesn't block me.